src/audio/pipeline/pipeline-stream.c

```diff
@@ -295,6 +396,15 @@ int pipeline_trigger(struct pipeline *p, struct comp_dev *host, int cmd)
 	case COMP_TRIGGER_PRE_RELEASE:
 	case COMP_TRIGGER_PRE_START:
 		/* Add all connected pipelines to the list and trigger them all */
+#if CONFIG_KCPS_DYNAMIC_CLOCK_CONTROL
+		/* setup walking ctx for removing consumption */
+		data.start = p->source_comp;
+		data.p = p;
+		walk_ctx.comp_func = add_pipeline_cps_consumption;
+
+		if(!clocks_handled)
+			ret = walk_ctx.comp_func(p->source_comp, NULL, &walk_ctx, PPL_DIR_DOWNSTREAM);
```

Comment:
if fails then should we do something?

Reply:
no fail here, any stop conditions means we are done with kcps calculations
(in new approach this walk_ctx func will only calculate kcps, clock change is done outside of this func)